src/storage/single_file_block_manager.cpp

```diff
@@ -159,6 +255,30 @@ SingleFileBlockManager::SingleFileBlockManager(AttachedDatabase &db, const strin
       iteration_count(0), options(options) {
 }
 
+void SingleFileBlockManager::LockEncryptionKey() {
+	auto &derived_key = options.encryption_options.derived_key;
+
+#if defined(_WIN32)
+	VirtualLock(const_cast<void *>(static_cast<const void *>(derived_key.data())), derived_key.size());
+#else
+	mlock(derived_key.data(), derived_key.size());
```

Comment:
Given we are going through lengths here not to page out the memory of the encryption key - I think we should not use a `std::string` to store the encryption key. It is very easy to accidentally copy over an `std::string` which would then negate this protection.

Can we perhaps rewrite the keys to use a separate class that is not copy constructible (and that calls `mlock`/`munlock` itself perhaps on memory that it is allocating to hold the key?).